src/widget/msApplicationLauncher.ts

```diff
@@ -41,11 +41,25 @@ export class MsApplicationLauncher extends St.Widget {
             style: 'padding:64px',
         });
         this.msWorkspace = msWorkspace;
-        this.add_style_class_name('surface-darker');
         this.appListContainer = new MsApplicationButtonContainer(
             this.msWorkspace
         );
-        this.initAppListContainer();
+
+        this.launcherChangedSignal = SignalHandle.connect(
+            Me.msThemeManager,
+            'show-application-launcher-changed',
+            () => {
+                if (Me.msThemeManager.showApplicationLauncher) {
+                    this.startAppListContainer();
+                } 
+                
+                if (!Me.msThemeManager.showApplicationLauncher) {
```

Comment:
I think this could be written more simply as `if / else`. Same for the identical condition in `restartAppListContainer`.

Reply:
When written as an if / else I was getting a strange bug that would bring up the Application Launcher.  Once it was switched this way the launcher remained disabled when it was disabled.